### ISSUE TITLE
Allow to compile to commonjs

### ISSUE THREAD

**Issue by codingtuba** [sveltejs/kit#5249]
### Describe the problem

When using svelte-kit, you aren't allowed to compile to commonjs.

### Describe the proposed solution

I would like the option to compile to commonjs.

### Importance

i cannot use SvelteKit without it

### Additional Information

Currently, namecheap is my hosting provider, and it doesn't support the import/export syntax (meaning that lots of people can't run svelte-kit apps for production)

**Comment by gtm-nayan** [comment]
Any particular reason why they don't allow ESM? And if so how do they enforce it?

Are you certain it's not just a bad package.json? Remember it needs to have a `"type": "module"` otherwise node will treat all js files in the project as CJS and error out if it encounters ESM syntax.

SvelteKit's output uses other modern JavaScript syntax (e.g. optional chaining), assuming you use the node adapter, if namecheap's node version doesn't support ESM it won't support those either.

**Comment by Rich-Harris** [comment]
Yep, I think this is a misunderstanding. It sounds like you just need to run a more up to date version of Node. If it's )really) not possible to do that, just compile the output of `adapter-node` (which I assume you're using) to CommonJS using esbuild or something.

Closing as there's no reason for SvelteKit to output commonjs
